.github/workflows/scripts/detect_ts_ignore.py

```
@@ -0,0 +1,92 @@
+                for line_num, line in enumerate(f, start=1):
+                    # Handle more variations of @ts-ignore
+                    if re.search(
+                        TS_IGNORE_PATTERN,
+                        line.strip(),
+                    ):
+                        print(
+                            f"❌ Error: '@ts-ignore' found in {file} "
+                            f"at line {line_num}"
+                        )
+                        logging.debug(
+                            f"Found @ts-ignore in line: {line.strip()}"
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)